### PR TITLE
Added option max_preserve_newlines in CSS beautifier

### LINKED ISSUE #1863: Option 'max_preserve_newlines' not working on beautify_css.js CSS Beautifier
# Description
> This is the default template for bug reports
> The option 'max_preserve_newline' is not working with CSS Beautifier. When the option 'preserve_newlines': true and 'max_preserve_newlines': 2, it is keeping all the newlines instead of 2 only.
> NOTE: 
> * Do not include screenshots! This library is a text processor, we need text inputs and outputs for debugging and fixing issues. 
> * Check the list of open issues before filing a new issue. 

# Input
The code looked like this before beautification:
```
p {





color: blue;
}
```

# Expected Output
The code should have looked like this after beautification:
```
p {

color: blue;
}
```

# Actual Output
The  code actually looked like this after beautification:
```
p {





color: blue;
}
```

# Steps to Reproduce
Use these options:
{
    'preserve_newlines': true,
    'max_preserve_newlines': 2,
}
It will keep all newlines instead of 2 only.

## Environment
OS: Chrome Browser

## Settings
Example:
```json
{
    'preserve_newlines': true,
    'max_preserve_newlines': 2,
}
```

### PR DESCRIPTION
**Pull request by JuanFML**
# Description
Added option max_preserve_newlines to CSS beautifier, as well tests with more lines, less lines and the same as the ones requested in the option.
- [x] Source branch in your fork has meaningful name (not `master`)

Fixes #1863: 

# Before Merge Checklist 
These items can be completed after PR is created.

(Check any items that are not applicable (NA) for this PR)

- [x] JavaScript implementation
- [x] Python implementation (NA if HTML beautifier)
- [x] Added Tests to data file(s)
- [ ] Added command-line option(s) (NA if
- [ ] README.md documents new feature/option(s)

### CODE REVIEW

File: js/src/css/beautifier.js [hunk 1]
```diff
@@ -92,11 +92,15 @@ Beautifier.prototype.eatString = function(endChars) {
 Beautifier.prototype.eatWhitespace = function(allowAtLeastOneNewLine) {
   var result = whitespaceChar.test(this._input.peek());
   var isFirstNewLine = true;
-
+  var newlines = this._options.max_preserve_newlines;
```
Comment (bitwiseman) [inline js/src/css/beautifier.js:95]: Delete `var isFirstNewLine = true;` and then 

```suggestion
  var newline_count = 0;
```

File: js/src/css/beautifier.js [hunk 2]
```diff
@@ -92,11 +92,15 @@ Beautifier.prototype.eatString = function(endChars) {
...
   while (whitespaceChar.test(this._input.peek())) {
     this._ch = this._input.next();
     if (allowAtLeastOneNewLine && this._ch === '\n') {
-      if (this._options.preserve_newlines || isFirstNewLine) {
+      if ((this._options.preserve_newlines || isFirstNewLine) && (this._options.max_preserve_newlines === 32786 || this._options.max_preserve_newlines === 0)) {
+        isFirstNewLine = false;
+        this._output.add_new_line(true);
+      } else if ((this._options.preserve_newlines || isFirstNewLine) && newlines > 0) {
+        newlines--;
```
Comment (bitwiseman) [inline js/src/css/beautifier.js:103]: max_preserve_newlines is already sanitized by the options object:

https://github.com/beautify-web/js-beautify/blob/aa9e365aed991080193c2bced411f132990c9d82/js/src/core/options.js#L44-47

```suggestion
      if (newline_count === 0 || newline_count < this._options.max_preserve_newlines) {
        newline_count++;
```

Also delete `isFirstNewLine = false;` below.

And implement same in python, of course.

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Thanks for contributing.  
Take a look at my suggestion and see if they seem right to you.

**Comment by bitwiseman** [comment]
@JuanFML  
Please remove 95e29b8 and the revert b14e245 from the branch for this PR.  Otherwise, #1860 will become an empty PR when this is merged: all the changes in it will already be in `master`.

File: js/src/css/beautifier.js (hunk 2 above)
Comment (JuanFML) [reply js/src/css/beautifier.js]: Yes I liked this way better, is way cleaner, thank you!

**Comment by JuanFML** [comment]
> @JuanFML
> Please remove [95e29b8](https://github.com/beautify-web/js-beautify/commit/95e29b8f9f08ccc1106de66df096ed332f93306a) and the revert [b14e245](https://github.com/beautify-web/js-beautify/commit/b14e2451edc5f2fd9de7368c3bbb68ab4395d7ad) from the branch for this PR. Otherwise, #1860 will become an empty PR when this is merged: all the changes in it will already be in `master`.

I think I already fix it, although I don't know how I can erase the commits at the top, but I could also close this and open a new one.

**Review by bitwiseman — APPROVED** [review]

**Pull request merged**